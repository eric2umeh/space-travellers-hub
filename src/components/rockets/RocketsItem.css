.rocket__info {
  display: flex;
  gap: 50px;
  justify-content: center;
  margin: 20px 0;
}

.rocket__info-image {
  width: 300px;
}

.rocket__info-details {
  display: flex;
  flex-direction: column;
  gap: 20px;
  justify-content: flex-start;
}

.rocket__info-btn {
  width: fit-content;
  padding: 10px 20px;
  background-color: #007aff;
  border-radius: 5px;
  color: white;
  border: none;
}

.cancel {
  background-color: transparent;
  color: gray;
  border: 1px solid gray;
}

.rocket__info-reserved {
  background-color: #18a2b8;
  color: white;
  border-radius: 5px;
  padding: 0 10px;
  display: inline-block;
}

/* Mobile View */
@media only screen and (max-width: 768px) {
  .rocket__info-image {
    width: 40%;
  }
}
