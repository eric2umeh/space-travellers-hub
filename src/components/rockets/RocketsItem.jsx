import PropTypes from 'prop-types';
import React from 'react';
import './RocketsItem.css';
import { useDispatch } from 'react-redux';
import { reserveRocket } from '../../redux/rockets/rocketsSlice';

const RocketsItem = ({ item }) => {
  const dispatch = useDispatch();
  const reserveHandler = () => {
    dispatch(reserveRocket(item.rocket_id));
  };

  return (
    <div className="rocket__info">
      <img
        src={item.rocket_images[0]}
        alt="Rocket"
        className="rocket__info-image"
      />
      <div className="rocket__info-details">
        <h4 className="rocket__info-heading">{item.rocket_name}</h4>
        <div className="rocket__info-reservation-inf">
          <p className="rocket__info-description">
            {item.reserved && (
              <span className="rocket__info-reserved">Reserved</span>
            )}
            {item.rocket_description}
          </p>
        </div>
        {item.reserved ? (
          <button
            type="submit"
            className="rocket__info-btn cancel"
            onClick={cancelHandler}
          >
            Cancel Rocket
          </button>
        ) : (
          <button
            type="submit"
            className="rocket__info-btn"
            onClick={reserveHandler}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  );
};

RocketsItem.propTypes = {
  item: PropTypes.string.isRequired,
};

export default RocketsItem;
