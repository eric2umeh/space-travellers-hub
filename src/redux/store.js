import { configureStore } from '@reduxjs/toolkit';
import rockets from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rockets.reducer,
  },
});

export default store;
