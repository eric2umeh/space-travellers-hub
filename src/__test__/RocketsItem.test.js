import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import RocketsItem from '../components/rocketsItem/RocketsItem';

it('test if the rocket component is rendering', () => {
  const item = {
    rocket_id: '1',
    rocket_name: 'Starship',
    rocket_description:
      'Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.',
    reserved: false,
    rocket_images: ['image1', 'image2'],
  };

  const test2 = renderer.create(
    <Provider store={store}>
      <RocketsItem item={item} />
    </Provider>,
  );

  expect(test2).toMatchSnapshot();
});
