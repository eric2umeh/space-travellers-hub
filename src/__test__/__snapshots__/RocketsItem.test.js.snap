// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if the rocket component is rendering 1`] = `
<div
  className="rocket__info"
>
  <img
    alt="Rocket"
    className="rocket__info-image"
    src="image1"
  />
  <div
    className="rocket__info-details"
  >
    <h4
      className="rocket__info-heading"
    >
      Starship
    </h4>
    <div
      className="rocket__info-reservation-inf"
    >
      <p
        className="rocket__info-description"
      >
        Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.
      </p>
    </div>
    <button
      className="rocket__info-btn"
      onClick={[Function]}
      type="submit"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
